_posts/2023-08-23-kafka-node-pools-storage-and-scheduling.md

```diff
@@ -0,0 +1,353 @@
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: pool-a
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 500Gi
+        class: fast
+  # ...
+---
+
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: pool-b
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 5Ti
+        class: slow
+  # ...
+```
+
+In the example above, `pool-a` uses fast and expensive storage of small capacity and `pool-b` uses a lot of slow and cheap storage.
+But while this setup might be useful in some situations, it is not always practical.
+Kafka lacks any advanced topic scheduling functionality.
+So you cannot instruct it that one topic should be placed on nodes with slow storage while other topics should be placed on nodes with fast storage.
+As a result, maintaining a setup like this might take a lot of effort.
+
+However, you can use this to easily change the storage your brokers are using and easily solve the scenarios described earlier.
+Shrinking the disks or changing the storage type can be done in a few simple steps.
+Let's imagine that we have an old Apache Kafka cluster using Amazon Elastic Block Storage GP2 volumes and we want to migrate it to use the newer and more performant GP3 storage.
+So we start with the following node pool:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: brokers
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  roles:
+    - broker
+  replicas: 3
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 1Ti
+        class: gp2-ebs
+  # ...
+```
+
+To migrate to the new storage, we have to first create a new node pool that uses the new storage type:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: brokers-gp3
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  roles:
+    - broker
+  replicas: 3
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 1Ti
+        class: gp3-ebs
+  # ...
+```
+
+Once the new brokers are deployed and ready, we can use Cruise Control and the `KafkaRebalance` resource to move everything from the old brokers to the new brokers:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels:
+    strimzi.io/cluster: my-cluster
+  annotations:
+    strimzi.io/rebalance-auto-approval: "true"
+spec:
+  mode: remove-brokers
+  brokers: [0, 1, 2] # Use the node IDs of the existing brokers
+  # ...
+```
+
+Now we have to wait until Cruise Control executes the rebalance for us.
```

Comment:
Can we include a snippet of what the finished state looks like from the CR? Or a `kubectl wait` comment or something just to make this _wait until_ step more concrete?